api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -381,47 +415,73 @@ private void validateAreaOfResearch(Profile profile) throws BadRequestException
    * object.
    *
    * @param updatedProfile
-   * @param prevProfile
+   * @param previousProfile
    * @throws BadRequestException
    */
   @VisibleForTesting
-  public void validateProfile(@Nonnull Profile updatedProfile, @Nullable Profile prevProfile)
+  public void validateProfile(
+      @Nonnull Profile updatedProfile, @Nullable Profile previousProfile, boolean userIsAdmin) {
+
+    final boolean isNewProfile = previousProfile == null;
+    final Diff diff = javers.compare(previousProfile, updatedProfile);
+
+    validateProfileForCorrectness(updatedProfile, isNewProfile, diff);
```

Comment:
Split into "this is a correct Profile" vs "user or admin can change this" for clarity